### ISSUE TITLE
Action not executable

### ISSUE THREAD

**Issue by jerrychee94** [pdmfc/nova-action-button#3]
Hi, in order to execute the action from field. We have to first define the resource action. If resource action was empty, we are unable to execute the action from field.

In ActionController.php :

**Comment by skoyah** [comment]
Hi @jerrychee94. Could you show an example so I can try to reproduce it?

Also, give me the info about your Laravel and Nova versions that you're currently using.

**Comment by jerrychee94** [comment]
```
    public function actions(Request $request)
    {
          return [];
   }


    public function fields(Request $request)
    {
          return [
            ActionButton::make('Add remark')->action((new AddRemark())->onlyOnDetail(), $this->id)
         ];
   }
```

Now you can try to execute the action on the field button. Nothing will happen.

Nova version : 2.11.1

**Comment by skoyah** [comment]
You are using it the wrong way. You should pass the action full class name as the first argument and not actually newing up the class like the example in the documentation.

**Comment by jerrychee94** [comment]
Tried with this but still not working unless I also include the same action in actions() method.
    
public function fields(Request $request)
    {
          return [
            ActionButton::make('Action')->action(AddRemark::class, $this->id),
         ];
   }

**Comment by jerrychee94** [comment]
When we execute the action, it will call 
 Nova.request({
                method: 'post',
                url: this.endpoint || `/nova-api/${this.resourceName}/action`,
                params: this.actionRequestQueryString,
                data: this.actionFormData(),
            })

and this is where it send post request to ActionController@store. However, if you trace along the methods, it will still depend on action() method from your resource.

See line 38 ActionRequest.php

A workaround solution I have in mind is to create trait and override 'resolveActions'. But I do found another issue in it. Maybe we discuss via chat ?

**Comment by skoyah** [comment]
You are right @jerrychee94 . In order to run the action, you have to include it on the `actions()` resource method.

I'll add this as a caveat to the documentation.

We can continue to discuss this issue here since might be helpful to other people.
